errors.js

```diff
@@ -7,16 +7,32 @@ class ERR_INVALID_ARG_TYPE extends TypeError {
   }
 }
 
+class ERR_INVALID_OPTION_VALUE extends Error {
```

Comment:
As @[cjihrig](https://github.com/cjihrig) suggests elsewhere, could se use:

```
> new errors.codes.ERR_INVALID_ARG_VALUE('options.foo.short', '--bar', 'must be a single character');
```

To avoid needing to introduce a new error to `errors.js`?

Reply:
Here's an early WIP with all the strict error cases updated to use `ERR_INVALID_ARG_VALUE`: https://github.com/pkgjs/parseargs/pull/74/commits/948ca9e7270664cb9c558f040db5a29c743964e1

I could use some suggestions for the error messages as the new message constraint of `ERR_INVALID_ARG_VALUE` and my current implementation doesn't read entirely accurate:
```js
// node invalid-string-option.js --foo
parseArgs({
  strict: true,
  foo: { type: 'string' }
})
// TypeError: The property --foo must be provided a value, Received undefined.
```
```js
// node invalid-boolean-option.js --foo=bar
parseArgs({
  strict: true,
  foo: { type: 'boolean' }
})
// TypeError: The property --foo does not expect a value, Received bar.
```

Really I'm hung up on the word `property` over `option` (e.g. `The option --foo must...`), but not sure if I'm overthinking it. Just want the message to be super clear to users.

Reply:
Similarly, the current [unknown option message](https://github.com/pkgjs/parseargs/blob/a2ea48de7ba3a5e48cfda50262cdf783bc672701/index.js#L92-L94) isn't entirely helpful to users e.g.
```sh
$ node unknown-option.js --foo

TypeError: The property option.foo must be configured, Received 'undefine'
```

Any suggestions to make this message more user-facing? e.g.
```
TypeError: The property --foo must be a known option, Received '--foo'
```

Reply:
> Any suggestions to make this message more user-facing? 

As a user I would expect the message to be "Unknown option --foo".

Reply:
Agreed, that was the original message before the [rework](https://github.com/pkgjs/parseargs/commit/948ca9e7270664cb9c558f040db5a29c743964e1) with the addition of dynamism for long and short options e.g. `Unknown long option --foo` and `Unknown short option -f`. In an effort to reduce friction merging into Node I'm trying to find an equally helpful message within the constraints of `ERR_INVALID_ARG_VALUE`.

Reply:
A quick comment: end-users do not know about the properties. They can see what they typed. So `--foo` and `-f` are  much more relevant than `foo`. And I think "option" is fine, do not need to specify "short" and "long", that is self evident.

(Disclaimer: I have not had a look yet at the constraints of `ERR_INVALID_ARG_VALUE` yet! Hopefully can go deeper tonight.)

Reply:
I don't think the format for `ERR_INVALID_ARG_VALUE` is a particularly good match for our use here where we are referring to options on the command line, rather than arguments to a javascript function. (Valiant try though!)

I see the node 18.x branch has updated code for the error which inserts property/argument depending on the name, but still not a good match for our option errors.

https://github.com/nodejs/node/blob/ec5a359ffd38b6f56a456a0b6a482b5167fbfce1/lib/internal/errors.js#L1246

Reply:
Reverted to custom error classes and updated the error messages per: https://github.com/pkgjs/parseargs/pull/74#issuecomment-1095569158

Examples based on the PR description prompts:
- Unknown option encountered
```
Error: Unknown option '--foo'
Error: Unknown option '-f'
```
- Option of `type:'string'` used like a boolean option e.g. lone `--string`
```
Error: Option '--foo <foo>' argument missing
Error: `Option '-f, --foo <foo>' argument missing
```
- Option of `type:'boolean'` used like a string option e.g. `--boolean=foo`
```
Error: Option '--foo' does not take an argument
Error: `Option '-f, --foo' does not take an argument
```

Reply:
Good point! That example is excessively long. I think I'll use `value` for consistency with the `parseArgs` results object.

Reply:
https://github.com/pkgjs/parseargs/pull/74/commits/870cffefd7d1cae28564c374b23eb21e6fb6c3c1